Legacy/BootPlatform/BlockIoDxe/BiosInt13.c

```diff
@@ -0,0 +1,1495 @@
+        // Toshiba 3 mode
+        //
+        case 0x0F:
+        //
+        // NEC 3 mode
+        //
+        case 0x10:
+          //
+          // Default Media
+          //
+          if ((Drive->Parameters.Flags & EDD_GEOMETRY_VALID) == EDD_GEOMETRY_VALID) {
+            Drive->MaxHead   = (UINT8)(Drive->Parameters.MaxHeads - 1);
+            Drive->MaxSector = (UINT8)Drive->Parameters.SectorsPerTrack;
+            ASSERT (Drive->MaxSector != 0);
+            Drive->MaxCylinder = (UINT16)(Drive->Parameters.MaxCylinders - 1);
+          } else {
+            Drive->MaxHead     = 0;
+            Drive->MaxSector   = 1;
+            Drive->MaxCylinder = 0;
+          }
+
+          break;
+
+        default:
+          //
+          // Unknown media type.
+          //
+          Drive->MaxHead     = 0;
+          Drive->MaxSector   = 1;
+          Drive->MaxCylinder = 0;
+          break;
+      }
+    }
+
+    Drive->Parameters.PhysicalSectors = (Drive->MaxHead + 1) * Drive->MaxSector * (Drive->MaxCylinder + 1);
+    Drive->Parameters.BytesPerSector  = 512;
+  }
+
+  //
+  // This data comes from the BIOS so it may not allways be valid
+  //  since the BIOS may reuse this buffer for future accesses
+  //
+  PointerMath        = EFI_SEGMENT (Drive->Parameters.Fdpt) << 4;
+  PointerMath       += EFI_OFFSET (Drive->Parameters.Fdpt);
+  Drive->FdptPointer = (VOID *)PointerMath;
+
+  return TRUE;
+}
+
+//
+// Block IO Routines
+//
+
+/**
+  Read BufferSize bytes from Lba into Buffer.
+
+  @param  This       Indicates a pointer to the calling context.
+  @param  MediaId    Id of the media, changes every time the media is replaced.
+  @param  Lba        The starting Logical Block Address to read from
+  @param  BufferSize Size of Buffer, must be a multiple of device block size.
+  @param  Buffer     A pointer to the destination buffer for the data. The caller is
+                     responsible for either having implicit or explicit ownership of the buffer.
+
+  @retval EFI_SUCCESS           The data was read correctly from the device.
+  @retval EFI_DEVICE_ERROR      The device reported an error while performing the read.
+  @retval EFI_NO_MEDIA          There is no media in the device.
+  @retval EFI_MEDIA_CHANGED     The MediaId does not matched the current device.
+  @retval EFI_BAD_BUFFER_SIZE   The Buffer was not a multiple of the block size of the device.
+  @retval EFI_INVALID_PARAMETER The read request contains LBAs that are not valid,
+                                or the buffer is not on proper alignment.
+
+**/
+EFI_STATUS
+EFIAPI
+Edd30BiosReadBlocks (
+  IN  EFI_BLOCK_IO_PROTOCOL  *This,
+  IN  UINT32                 MediaId,
+  IN  EFI_LBA                Lba,
+  IN  UINTN                  BufferSize,
+  OUT VOID                   *Buffer
+  )
+{
+  EFI_BLOCK_IO_MEDIA         *Media;
+  BIOS_BLOCK_IO_DEV          *BiosBlockIoDev;
+  EDD_DEVICE_ADDRESS_PACKET  *AddressPacket;
+  //
+  // I exist only for readability
+  //
```

Comment:
If I have not misunderstood, this and the three similar below are to get round uncrustify insisting on matching tabbing for all local variables.

Apologies, but I would disagree that having to stop and parse this while reading the code improves readability, and suspect/suggest (to the extent that I have any reasonable say!) that it would be better/simpler (and probably more readable overall) to remove these and do what uncrustify wants.

Reply:
Removed, leftovers from EDK2 version of the driver.